### PR TITLE
Gather mount facts, fallback for when multiproc is not feasable

### LINKED ISSUE #83746: Inaccessible `/dev/shm` causes the `ansible_mounts` fact not to be collected
### Summary

I was doing some initial OS bootstrapping automation and encountered an issue where I couldn't get the mount facts available on the target node. A little bit of debugging revealed a shared memory permission problem. I realized that there's probably even more facts that aren't being populated.

The thread pool creation here https://github.com/ansible/ansible/blob/59ca05b/lib/ansible/module_utils/facts/hardware/linux.py#L580 raises a `PermissionError` which leaks into the calling function and doesn't get captured @ https://github.com/ansible/ansible/blob/59ca05b/lib/ansible/module_utils/facts/hardware/linux.py#L100 so it interrupts the `populate()` method's flow too. I suppose it gets suppressed somewhere up the stack.

The proposed solution would be to issue a warning on such an error and attempt a fallback strategy.

I'm cleaning out my browser tabs and I don't have the exact traceback from when I was looking into this. But it was similar to https://stackoverflow.com/q/2009278/595220.

The underlying `_multiprocessing.SemLock()` creation is what's failing due to its reliance on the shared memory availability. Here's some trail I was following in CPython: https://github.com/python/cpython/blob/bc94cf7e2/Lib/multiprocessing/managers.py#L1267 / https://github.com/python/cpython/blob/bc94cf7e2/Lib/multiprocessing/shared_memory.py#L95 / https://github.com/python/cpython/blob/bc94cf7e2/Modules/_multiprocessing/posixshmem.c#L31-L74 / https://github.com/python/cpython/blob/main/Modules/_multiprocessing/clinic/posixshmem.c.h#L5-L43.

### Issue Type

Bug Report

### Component Name

`/lib/ansible/module_utils/facts/hardware/linux.py`

### Ansible Version

```console
devel
```

### Configuration

```console
# if using a version older than ansible-core 2.12 you should omit the '-t all'
$ ansible-config dump --only-changed -t all
```

### OS / Environment

Fedora LiveUSB, I think.

### Steps to Reproduce

I don't have the exact steps, but it seems like `sudo chmod u=,g=,o= /dev/shm` on the target node should be enough, and then run something like `ansible -m ansible.builtin.setup -a 'filter=ansible_mounts'` without “become”. Adding `-b` or `sudo chmod u=rwx,g=rwx,o=rwx /dev/shm` appears to be a workaround.

### Expected Results

Not sure. At least a warning printed out would be nice. Ideally, maybe some fallback mechanism could be implemented.

### Actual Results

```console
The list of gathered mounts is empty. There is no hint in the program output mentioning there might be a runtime problem.
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by bcoca**
fixes #83746

##### ISSUE TYPE

- Bugfix Pull Request

### CODE REVIEW

File: lib/ansible/module_utils/facts/hardware/linux.py [hunk 1]
```diff
@@ -577,7 +582,13 @@ def get_mount_facts(self):
 
         # start threads to query each mount
         results = {}
-        pool = ThreadPool(processes=min(len(mtab_entries), cpu_count()))
+        pool = None
+        try:
+            pool = ThreadPool(processes=min(len(mtab_entries), cpu_count()))
+        except Exception as e:
```
Comment (webknjaz) [inline lib/ansible/module_utils/facts/hardware/linux.py]: Can we handle `PermissionError` explicitly? Catch-alls are bad for debugging.

Comment (bcoca) [reply lib/ansible/module_utils/facts/hardware/linux.py]: It is not the only possible error, was keeping exception until i could compile a full list

File: test/integration/targets/gathering_facts/lib/multriprocessing/pool/__init__.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from __future__ import annotations
+
+
+class ThreadPool:
+
+    def init(self, *args, **kwargs):
```
Comment (webknjaz) [inline test/integration/targets/gathering_facts/lib/multriprocessing/pool/__init__.py]: Did you mean?
```suggestion
    def __init__(self, *args, **kwargs):
```

Comment (bcoca) [reply test/integration/targets/gathering_facts/lib/multriprocessing/pool/__init__.py]: i dumb

**Review by Apelsam — APPROVED** [review]

**Review by Apelsam — APPROVED** [review]

**Pull request merged**